src/rule_dsl.jl

```diff
@@ -243,7 +248,8 @@ function (r::RuleSet)(term; depth=typemax(Int))
                 # this rule doesn't apply
                 continue
             else
-                return r(expr′, depth=getdepth(rules[i])) # levels touched
+                expr = r(expr′, depth=getdepth(rules[i]))# levels touched
+                r.applyall || return expr
```

Comment:
So does this say if there's one change go to the parent `RuleSet`? Seems like the same amount of info as recurse :) and sure enough in your code you either set recurse or this and not both. Why not use `!r.recurse` ? Better yet, make it an optional state of this function!

Reply:
Yeah, it should be an option for the function instead of the ruleset. It is different than `recurse`. If you have `RuleSet([r1, r2])` and `r1` always matches, you'll never use `r2` if you don't turn on `applyall`.

Reply:
no matter what `applyall` is, this line just leads to `return expr` here or below. Am I reading it wrong?